### DISCUSSION TITLE
Conditional saving/storing in a jitted function

### DISCUSSION THREAD

**Discussion by andremfreitas** [jax-ml/jax#20592] [Q&A] [answered]
Hi,

What is the optimal approach when we want to save some result every, for example, 10 steps, of a numerical solver ? `jax.lax.cond` seems not to be ideal because the true and false branches need to return the same type.  

PS: This numerical solver is called iteratively via a `jax.lax.fori_loop` or a `jax.lax.scan`

Thanks,
Andre

**Comment by jakevdp** [comment, marked as answer]
Saving a value is a side-effect, so you cannot do it directly in a `jit`-compiled function. But you should be able to use [`jax.experimental.io_callback`](https://jax.readthedocs.io/en/latest/_autosummary/jax.experimental.io_callback.html) for this; see a discussion at [External Callbacks in JAX](https://jax.readthedocs.io/en/latest/notebooks/external_callbacks.html).

**Reply by andremfreitas** [reply]
Thanks! This is exactly what I was looking for :)
